build.sbt

```diff
@@ -33,26 +33,82 @@ inThisBuild(
   )
 )
 
-scalacOptions ++= Seq(
-  "-deprecation",
-  "-Wunused:all",
-  "-feature",
-  "-unchecked",
-  "-Xcheck-macros",
-  "-Xprint-suspension",
-  "-Xsemanticdb",
-  "-Yexplicit-nulls",
-  "-Ysafe-init",
-  "-source:future",
-  "-Ykind-projector",
-  "-Vprofile"
-)
+lazy val slinc = project
+  .in(file("slinc"))
+  .settings(
+    name := "slinc",
+    scalacOptions ++= Seq(
+      "-deprecation",
+      "-Wunused:all",
+      "-feature",
+      "-unchecked",
+      "-Xcheck-macros",
+      "-Xprint-suspension",
+      "-Xsemanticdb",
+      "-Yexplicit-nulls",
+      "-Ysafe-init",
+      "-source:future",
+      "-Ykind-projector",
+      "-Vprofile"
+    ),
+    libraryDependencies += "org.scalameta" %% "munit" % "1.0.0-M10" % Test,
+    libraryDependencies += "org.scalameta" %% "munit-scalacheck" % "1.0.0-M10" % Test,
+    libraryDependencies += "org.scala-lang" %% "scala3-staging" % scalaVersion.value,
+    Compile / doc / scalacOptions ++= Seq("-siteroot", "docs"),
+    mimaPreviousArtifacts := previousStableVersion.value
+      .map(organization.value %% moduleName.value % _)
+      .toSet
+  )
 
-libraryDependencies += "org.scalameta" %% "munit" % "1.0.0-M10" % Test
-libraryDependencies += "org.scalameta" %% "munit-scalacheck" % "1.0.0-M10" % Test
-libraryDependencies += "org.scala-lang" %% "scala3-staging" % scalaVersion.value
+lazy val copyNative = TaskKey[Set[File]]("copy shared lib")
 
-Compile / doc / scalacOptions ++= Seq("-siteroot", "docs")
-mimaPreviousArtifacts := previousStableVersion.value
-  .map(organization.value %% moduleName.value % _)
-  .toSet
+lazy val examples = project
+  .in(file("slinc-examples"))
+  .dependsOn(slinc)
+  .settings(
+    publishTo := None,
+    publishLocal := Def.task(()),
+    copyNative := {
```

Comment:
This copies a shared lib from examples-native project to load it from resource.
https://github.com/scala-interop/slinc/pull/222/files#diff-d7b7c9d5ea6a2ac0cbbd2da77262f2b3074469d52c44ad9dd3a3dd2b16f00ebbR7

` Compile / resourceGenerators` is invoked at `run` and `package`.

> Executing run (or package, not compile) will add a file demo to resourceManaged, which is target/scala-*/resource_managed". By default, generated resources are not included in the packaged source artifact. To do so, add them as you would other mappings. See [Adding files to a package](https://www.scala-sbt.org/1.x/docs/Howto-Package.html#modify-package-contents).
> https://www.scala-sbt.org/1.x/docs/Howto-Generating-Files.html